hashednode.go

```diff
@@ -31,14 +31,37 @@ import (
 )
 
 type HashedNode struct {
-	commitment  []byte
+	commitment  SerializedPointCompressed
+	serialized  SerializedPoint
 	cachedPoint *Point
 }
 
+func NewHashedNode(buf SerializedPoint) *HashedNode {
+	var h HashedNode
+	h.serialized = buf
+
+	return &h
+}
```

Comment:
Before, we didn't have a constructor, and that felt like the wild-west on how this _HashedNode_ was created.
I wanted to centralize the logic in a clear place, and that place is here.

Also, as explained before, it is clear that the way to create a _HashedNode_ is with the serialized bytes (i.e: uncompressed form of the point).